src/megaclient.cpp

```diff
@@ -13657,6 +13657,26 @@ bool MegaClient::syncup(LocalNode* l, dstime* nds, size_t& parentPending)
             continue;
         }
 
+        // Make sure the local node is actually included.
+        {
+            LocalPath localPath = ll->getLocalPath();
+
+            if (!app->sync_syncable(ll->sync, localname.c_str(), localPath))
```

Comment:
thinking-out-loud:
MegaApiImpl::sync_syncable would check for size restrictions, but also for those based on localpath in MegaApiImpl::is_syncable, unlocking and re-locking mutex.
I wonder if that is costly enough so that we should consider avoiding the later, since the checks based on path exclusions would evaluate true at this point. Wouldn't they?